content/en/blog/_posts/2023-01-15-Security-Bahavior-Analysis/index.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: blog
+title: "All Microservices Are Vulnerable"
```

Comment:
I don't agree with this title. We should build consensus that the Kubernetes project is happy to endorse this message.

A less controversial message is to design _as if_ your code is riddled with risk, but not to actually assert that it truly is.

Reply:
(I don't want people to think that we're announcing a Kubernetes vulnerability, for example).

Reply:
@sftim, 
The title and content of this blog were discussed with the 6 reviewers from the Security-Tooling WG who reviewed this blog during a joint meeting we had to discuss this blog. 

We discussed the main message and the reasoning for this title. The title was changed from "Vulnerable Microservices" as a result to this one. 

I stand behind this title 100% since it is exactly what we do want people to understand. That the microservices deployed are all vulnerable and should take this into account. The assumption that the deployed microservices are not vulnerable is exactly what we are trying to root out in this article. No, in practice, it is impossible to build a weakness-free microservice that will not be vulnerable.

This title does not say Kubernetes is vulnerable  - it says that the deployed microservices are all vulnerable and it is about time the community will start accepting this reality and start acting accordingly.

Reply:
/hold

Let's make sure the project as a whole is happy with the comms side of things before we ship it. It _is_ a contentious title.

Reply:
Aside: https://github.com/kubernetes/community/ doesn't list a WG Security Tooling.

Reply:
You might mean https://github.com/kubernetes/community/tree/master/sig-security#security-tooling ?

Reply:
RIght

Reply:
Please let me know where should it be discussed beyond the discussion we already had in the sig-security-tooling WG meeting. I will try to make the case for accepting the post and its message. I thought we already passed this phase after the sig-security-tooling WG review, but I now understand that we did not.

I think it is great to get this posted in the k8s community blog (as it is very applicable to the community users and also important to the community developers) and be happy to explain the security aspects to whoever wishes to listen.  

If this blog is not accepted, np and I will publish it elsewhere.

Reply:
> The title and content of this blog were discussed with the 6 reviewers from the Security-Tooling WG who reviewed it during a joint meeting we had to discuss this blog.
...
> Please let me know where should it be discussed beyond the discussion we already had in the sig-security-tooling WG meeting. I will try to make the case for accepting the post and its message. I thought we already passed this phase after the sig-security-tooling WG review, but I now understand that we did not.

SIG Docs' Blog subproject manages the review process for blogs. Discussion on the content and title can continue on this PR. Folks from SIG Security can continue with the technical review on this PR and provide a `/lgtm` to confirm the technical review for this blog post.

Reply:
From @sftim 
> Let's make sure the project as a whole is happy with the comms side of things before we ship it. It is a contentious title.
...
From @davidhadas 
> I stand behind this title 100% since it is exactly what we do want people to understand. That the microservices deployed are all vulnerable and should take this into account. The assumption that the deployed microservices are not vulnerable is exactly what we are trying to root out in this article. No, in practice, it is impossible to build a weakness-free microservice that will not be vulnerable.

Let's tune the title so that it's not contentious and it conveys the author's message/intention.

@davidhadas, since deployed microservices are vulnerable (I agree with you on this), imo it's why a zero trust model should be applied everywhere, and this blog introduces _security behavior_, how about:

- Microservices Are Vulnerable — Analyze Their _Security Behavior_
- Yes, Your Microservices Are Vulnerable or Your Microservices Are Vulnerable
- Analyze the _Security Behavior_ of Microservices in Kubernetes

I'm not the best writer but what I'm trying to do above is to keep the impact of "All Microservices are Vulnerable" but defer the context of "microservices"  a bit

Reply:
Agree with Rey above.

Reply:
Changed to "All Your Microservices Are Vulnerable"
If will not be accepted, will change to "Your Microservices Are Vulnerable"